Gems/EMotionFX/Code/Source/Integration/Editor/Components/EditorActorComponent.cpp

```diff
@@ -124,12 +124,12 @@ namespace EMotionFX
                         ->Attribute(AZ::Edit::Attributes::AutoExpand, true)
                         ->DataElement(0, &EditorActorComponent::m_renderCharacter,
                             "Draw character", "Toggles rendering of character mesh.")
-                        ->Attribute(AZ::Edit::Attributes::ChangeNotify, &EditorActorComponent::OnDebugDrawFlagChanged)
+                        ->Attribute(AZ::Edit::Attributes::ChangeNotify, &EditorActorComponent::OnRenderFlagChanged)
                         ->DataElement(0, &EditorActorComponent::m_renderSkeleton,
                             "Draw skeleton", "Toggles rendering of skeleton.")
-                        ->Attribute(AZ::Edit::Attributes::ChangeNotify, &EditorActorComponent::OnDebugDrawFlagChanged)
+                        ->Attribute(AZ::Edit::Attributes::ChangeNotify, &EditorActorComponent::OnRenderFlagChanged)
                         ->DataElement(0, &EditorActorComponent::m_renderBounds, "Draw bounds", "Toggles rendering of world space bounding boxes.")
-                        ->Attribute(AZ::Edit::Attributes::ChangeNotify, &EditorActorComponent::OnDebugDrawFlagChanged)
+                        ->Attribute(AZ::Edit::Attributes::ChangeNotify, &EditorActorComponent::OnRenderFlagChanged)
```

Comment:
Does this mean that the render flags aren't exposed to the UI yet while they are serialized for the runtime component? Can we sync these?

Reply:
the runtime component will just have the renderFlags, not m_renderCharacter, m_renderBound and m_renderSkeleton. So there is nothing to sync for runtime component. For the editorActorComponent, since there is no good way to only expose the settings only for those three boolean (or perhaps there is but I didn't find it), those three boolean value are there and UpdateRenderFlags function is used to sync them.